spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -189,6 +190,26 @@ public void setPublishCallback(ListenableFutureCallback<String> publishCallback)
 		this.publishCallback = publishCallback;
 	}
 
+	public Expression getTopicExpression() {
+		return this.topicExpression;
+	}
+
+	/**
+	 * Set the SpEL expression for the topic this adapter sends messages to.
+	 * @param topicExpression SpEL expression representing the topic name
+	 */
+	public void setTopicExpression(Expression topicExpression) {
```

Comment:
Consider also to add ` setTopicExpressionString(String topicExpression)` and use a `EXPRESSION_PARSER` from the super class to parse provided String to the `Expression`.

This is pretty convenient for JavaConfig